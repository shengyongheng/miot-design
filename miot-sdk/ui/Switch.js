import { Switch } from 'mhui-rn';
export default Switch;