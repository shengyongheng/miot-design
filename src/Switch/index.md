---
title: Switch
order: 1
group:
  path: '/basic'
  title: 基础组件
  order: 1
nav:
  title: '组件'
  path: /components
---

# Button 按钮

## 简介

Button 组件是一个基础的交互元素，用于触发操作。

## 用法

```jsx
import React from 'react';
import { Text } from 'react-native';
import { Switch } from 'mhui-rn';

export default () => (
  <>
    <Text>基础使用</Text>
    <Switch style={{ width: 60, height: 30 }} />
    <Text>尺寸</Text>
    <Switch style={{ width: 20, height: 10 }} />
    <Text>选中</Text>
    <Switch
      style={{ width: 20, height: 10 }}
      onTintColor="red"
      tintColor="blue"
      value={true}
    />
  </>
);
```

## API

| 属性  | 类型    | 默认值    | 说明                                           |
| ----- | ------- | --------- | ---------------------------------------------- |
| type  | string  | 'default' | 按钮类型，可选值为 'primary', 'dashed', 'link' |
| size  | string  | 'middle'  | 按钮尺寸，可选值为 'large', 'middle', 'small'  |
| shape | string  | 'default' | 按钮形状，可选值为 'circle', 'round'           |
| value | boolean | false     | 指定当前是否选中                               |
| ...   | ...     | ...       | ...                                            |
