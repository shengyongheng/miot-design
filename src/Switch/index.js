// @ts-ignore
import Switch from 'miot/ui/Switch';
import React, { useState } from 'react';

const App = () => {
  const [value, setValue] = useState(false);

  return (
    <Switch
      style={{ width: 20, height: 10 }}
      onTintColor="red"
      tintColor="blue"
      value={value}
      onValueChange={(value) => setValue(value)}
    />
  );
};

export default App;
