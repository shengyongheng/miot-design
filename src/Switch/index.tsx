// @ts-ignore
import Switch from 'miot/ui/Switch';
import React from 'react';

const Index: React.FC<any> = () => {
  return (
    <Switch
      style={{ width: 20, height: 10 }}
      onTintColor="red"
      tintColor="blue"
      value={true}
      disabled={false}
    />
  );
};

export default Index;
