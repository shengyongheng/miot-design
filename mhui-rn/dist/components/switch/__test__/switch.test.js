import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Switch from '../Switch';

configure({ adapter: new Adapter() });

const switchProps = {
  value: false,
  onValueChange(value: boolean): void {
    // eslint-disable-next-line no-console
    console.log(value);
  },
};

describe('Switch', () => {
  describe('render', () => {
    it('render Switch', () => {
      const tree = renderer.create(
        <Switch value={switchProps.value} onValueChange={switchProps.onValueChange} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Interaction', () => {
    describe('onValueChange', () => {
      const mockOnPress = jest.fn();
      let instance1;
      beforeEach(() => {
        instance1 = shallow(<Switch
          value={false}
          onValueChange={mockOnPress}
        />).instance();
        jest.clearAllMocks();
      });
      it('should call onValueChange', () => {
        instance1.handleValueChange();
        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
      });
      it('should toggle state', () => {
        instance1.handleValueChange();
        expect(instance1.state.value).toBe(true);
        instance1.handleValueChange();
        expect(instance1.state.value).toBe(false);
      });
    });
  });
});
