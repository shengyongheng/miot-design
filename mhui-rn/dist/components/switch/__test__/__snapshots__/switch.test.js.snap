// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch render render Switch 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      Object {
        "opacity": 1,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F0F0F0",
        "borderColor": "rgba(0,0,0,0.1)",
        "borderRadius": 12,
        "borderWidth": 0.5,
        "height": 24,
        "justifyContent": "center",
        "opacity": 1,
        "width": 44,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "rgba(0,0,0,0.1)",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "height": 16,
          "margin": 4,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 16,
        }
      }
    />
  </View>
</View>
`;
